sdk/eventhub/azure-eventhubs/azure/eventhub/aio/producer_async.py

```diff
@@ -301,8 +306,25 @@ def _set_partition_key(event_datas, partition_key):
             ed._set_partition_key(partition_key)
             yield ed
 
+    async def create_batch(self, max_size=None, partition_key=None):
```

Comment:
I wonder if this function needs to be async.
I think it might be better to raise an error if this is called before the link is open, rather than have it open the connection.
Then the function doesn't need to be async and can simply return the Batch object.

Reply:
As we're not exposing the **producer.open** api to user, if users want to use create_batch before they send, we need to open the connection for them internally to get the right peer link message size.